### PR TITLE
Update JDK to 17.

### PR DESCRIPTION
**Pull request by armogur**
*Issue #, if available:* 155

*Description of changes:*
Switch from javax.jms.* to jakarta.jms.* namespace.
Update testing libraries and maven plugins.
Update JDK to 17.
Code cleanup, fixed typos in comments, used newer assertions in test code.

### CODE REVIEW

**Comment by AlexRosier** [comment]
Can somebody from AWS / with write access please look at this. This would enable us to switch to spring boot 3

**Comment by armogur** [comment]
> Can somebody from AWS / with write access please look at this. This would enable us to switch to spring boot 3

@AlexRosier at this moment no one will receive your message as there is no other participant, maybe you should mention someone from AWS java team.

**Comment by volkanto** [comment]
@ziyanli-amazon fyi ☝️

File: src/main/java/com/amazon/sqs/javamessaging/SQSConnection.java [hunk 1]
```diff
@@ -179,26 +179,26 @@ public QueueSession createQueueSession(boolean transacted, int acknowledgeMode)
      *             transaction and acknowledge mode.
      */
     @Override
-    public Session createSession(boolean transacted, int acknowledgeMode) throws JMSException {
+    public Session createSession(boolean transacted, int sessionMode) throws JMSException {
```
Comment (ziyanli-amazon) [inline src/main/java/com/amazon/sqs/javamessaging/SQSConnection.java:182]: Any particular reason to change the variable naming here?

Comment (armogur) [reply src/main/java/com/amazon/sqs/javamessaging/SQSConnection.java:182]: ohhh, right, something went sideways from my side, just checked the signature of the method, I will revert it back tomorrow.

File: src/test/java/com/amazon/sqs/javamessaging/message/SQSObjectMessageTest.java [hunk 2]
```diff
@@ -14,22 +14,16 @@
...
 
-import javax.jms.JMSException;
-import javax.jms.ObjectMessage;
-
-import org.junit.Assert;
-import org.junit.Test;
-
-import com.amazon.sqs.javamessaging.message.SQSObjectMessage;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (ziyanli-amazon) [inline src/test/java/com/amazon/sqs/javamessaging/message/SQSObjectMessageTest.java:26]: Please do implicit import like 
`import static org.junit.jupiter.api.Assertions.assertNotNull`

File: src/test/java/com/amazon/sqs/javamessaging/message/SQSObjectMessageTest.java [hunk 3]
```diff
@@ -70,20 +62,20 @@ public void testCreateMessageWithObject() throws JMSException {
...
     @Test
     public void testObjectSerialization() throws JMSException {
-        Map<String, String> expectedPayload = new HashMap<String, String>();
+        Map<String, String> expectedPayload = new HashMap<>();
         expectedPayload.put("testKey", "testValue");
 
         SQSObjectMessage sqsObjectMessage = new SQSObjectMessage();
 
-        String serialized = sqsObjectMessage.serialize((Serializable) expectedPayload);
+        String serialized = SQSObjectMessage.serialize((Serializable) expectedPayload);
```
Comment (ziyanli-amazon) [inline src/test/java/com/amazon/sqs/javamessaging/message/SQSObjectMessageTest.java:70]: Should this be
`String serialized = sqsObjectMessage.serialize((Serializable) expectedPayload);` instead?

File: src/test/java/com/amazon/sqs/javamessaging/message/SQSMessageTest.java [hunk 4]
```diff
@@ -14,32 +14,26 @@
...
+import org.junit.jupiter.api.Test;
 import software.amazon.awssdk.services.sqs.model.MessageAttributeValue;
 import software.amazon.awssdk.services.sqs.model.MessageSystemAttributeName;
 
 import java.util.*;
 
+import static com.amazon.sqs.javamessaging.SQSMessagingClientConstants.APPROXIMATE_RECEIVE_COUNT;
+import static com.amazon.sqs.javamessaging.SQSMessagingClientConstants.JMSX_DELIVERY_COUNT;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (ziyanli-amazon) [inline src/test/java/com/amazon/sqs/javamessaging/message/SQSMessageTest.java:34]: Please do explicit import.

File: src/test/java/com/amazon/sqs/javamessaging/message/SQSMessageTest.java [hunk 5]
```diff
@@ -14,32 +14,26 @@
...
+import static org.mockito.Mockito.*;
```
Comment (ziyanli-amazon) [inline src/test/java/com/amazon/sqs/javamessaging/message/SQSMessageTest.java:35]: same as above

File: src/test/java/com/amazon/sqs/javamessaging/message/SQSBytesMessageTest.java [hunk 6]
```diff
@@ -14,56 +14,34 @@
...
 import com.amazon.sqs.javamessaging.SQSSession;
-
+import jakarta.jms.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 import software.amazon.awssdk.services.sqs.model.Message;
 import software.amazon.awssdk.services.sqs.model.MessageSystemAttributeName;
 import software.amazon.awssdk.utils.BinaryUtils;
 
+import java.io.*;
```
Comment (ziyanli-amazon) [inline src/test/java/com/amazon/sqs/javamessaging/message/SQSBytesMessageTest.java:26]: https://stackoverflow.com/questions/187453/import-package-vs-import-package-specifictype

File: src/main/java/com/amazon/sqs/javamessaging/AmazonSQSMessagingClientWrapper.java [hunk 7]
```diff
@@ -55,39 +37,30 @@
...
-    }
+
+    /**
+     * List of exceptions that can classified as security. These exceptions
+     * are not thrown during connection-set-up rather after the service
+     * calls of the <code>SqsClient</code>.
+     */
+    private static final Set<String> SECURITY_EXCEPTION_ERROR_CODES = Set.of("MissingClientTokenId",
+            "InvalidClientTokenId", "MissingAuthenticationToken", "AccessDenied");
+    ;
```
Comment (ziyanli-amazon) [inline src/main/java/com/amazon/sqs/javamessaging/AmazonSQSMessagingClientWrapper.java:48]: extra semicolon?

File: src/main/java/com/amazon/sqs/javamessaging/PrefetchManager.java [hunk 8]
```diff
@@ -25,19 +25,19 @@ public interface PrefetchManager {
      * messageQueue when user calls for receive or message listener onMessage is
      * called.
      */
-    public void messageDispatched();
+    void messageDispatched();
```
Comment (ziyanli-amazon) [inline src/main/java/com/amazon/sqs/javamessaging/PrefetchManager.java:28]: It is still better to retain the class decoration so to be conformed with the package convention

File: src/main/java/com/amazon/sqs/javamessaging/SQSConnection.java (hunk 1 above)
Comment (ziyanli-amazon) [reply src/main/java/com/amazon/sqs/javamessaging/SQSConnection.java:182]: @armogur thank you! And also be aware that, you should point your fork to merge awslabs:v2 branch as opposed to the master branch. The master branch is intended for java v1 and v2 is for java v2. I will work on upgrading for java v1 while you already took care of the v2. I left some minor comments there, the majorities are format related. Please help us retain a good and uniformed package code style and convention. thanks a lot.

File: src/main/java/com/amazon/sqs/javamessaging/SQSSession.java [hunk 9]
```diff
@@ -731,114 +687,149 @@ void waitForCallbackComplete() {
...
-        }
-        
-        public MessageListener getMessageListener() {
-            return messageListener;
-        }
-        
-        public MessageManager getMessageManager() {
-            return messageManager;
-        }
+    record CallbackEntry(MessageListener messageListener, MessageManager messageManager) {
```
Comment (ziyanli-amazon) [inline src/main/java/com/amazon/sqs/javamessaging/SQSSession.java:830]: what is this method doing?

File: src/main/java/com/amazon/sqs/javamessaging/SQSConnection.java (hunk 1 above)
Comment (armogur) [reply src/main/java/com/amazon/sqs/javamessaging/SQSConnection.java]: fixed, please check.

File: src/test/java/com/amazon/sqs/javamessaging/message/SQSMessageTest.java (hunk 4 above)
Comment (armogur) [reply src/test/java/com/amazon/sqs/javamessaging/message/SQSMessageTest.java]: Fixed in every file, please check.

File: src/test/java/com/amazon/sqs/javamessaging/message/SQSMessageTest.java (hunk 5 above)
Comment (armogur) [reply src/test/java/com/amazon/sqs/javamessaging/message/SQSMessageTest.java]: Fixed in every file, please check.

File: src/test/java/com/amazon/sqs/javamessaging/message/SQSBytesMessageTest.java (hunk 6 above)
Comment (armogur) [reply src/test/java/com/amazon/sqs/javamessaging/message/SQSBytesMessageTest.java]: Fixed in every file, please check.

File: src/main/java/com/amazon/sqs/javamessaging/SQSSession.java (hunk 9 above)
Comment (armogur) [reply src/main/java/com/amazon/sqs/javamessaging/SQSSession.java:854]: I don't get it what you mean. The
```java
record CallbackEntry(MessageListener messageListener, MessageManager messageManager) {
}
```
record class is the replacement of [SQSSession.java](https://github.com/awslabs/amazon-sqs-java-messaging-lib/blob/master/src/main/java/com/amazon/sqs/javamessaging/SQSSession.java#L824-L840)

File: src/main/java/com/amazon/sqs/javamessaging/AmazonSQSMessagingClientWrapper.java (hunk 7 above)
Comment (armogur) [reply src/main/java/com/amazon/sqs/javamessaging/AmazonSQSMessagingClientWrapper.java]: fixed, please check

File: pom.xml [hunk 10]
```diff
@@ -1,207 +1,223 @@
...
+                        <goals>
+                            <goal>jar-no-fork</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <version>3.4.1</version>
```
Comment (ziyanli-amazon) [inline pom.xml:158]: I'm curious, did you see a lot of `[WARNING]` after upgrading the javadoc version here without having the `<doclint>none</doclint>` field?

**Review by ziyanli-amazon — APPROVED** [review]
Thanks for doing this @armogur. I have also published a PR to java v1, awaiting for internal review.

**Pull request merged**